File: sdkwork-ai-sdk-swift/Sources/API/Rerank2Api.swift
import Foundation

public class Rerank2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Rerank documents
    public func rerank(body: RerankRequest) async throws -> RerankResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/rerank"), body: body)
        return response as? RerankResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/Rerank2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Rerank2Api
    {
        private readonly HttpClient _client;

        public Rerank2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Rerank documents
        /// </summary>
        public async Task<RerankResponse?> RerankAsync(RerankRequest body)
        {
            return await _client.PostAsync<RerankResponse>(ApiPaths.AiPath("/v1/rerank"), body);
        }
    }
}
